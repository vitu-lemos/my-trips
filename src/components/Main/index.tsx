const Main = ({ title = 'React AvanÃ§ado' }) => <h1>{title}</h1>
export default Main
