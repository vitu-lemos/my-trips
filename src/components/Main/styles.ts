import styled from 'styled-components'

export const Wrapper = styled.div`
  height: 100%;
  padding: 3rem;
  background-color: #06092b;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: #fff;
`
export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`
export const Title = styled.h1`
  font-size: 2.5rem;
  font-weight: bold;
`
export const Description = styled.h2`
  font-size: 2rem;
  margin-bottom: 3rem;
  font-weight: 400;
`
export const Ilustration = styled.img`
  width: min(30rem, 100%);
`
