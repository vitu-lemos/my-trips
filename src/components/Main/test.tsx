import { screen, render } from '@testing-library/react'

import Main from './index'

describe('<Main>React avançado</Main>', () => {
  it('should render a header', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: 'React Avançado' })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
