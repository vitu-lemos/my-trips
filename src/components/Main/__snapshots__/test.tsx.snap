// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main>React avançado</Main> should render a header 1`] = `
.c0 {
  height: 100%;
  padding: 3rem;
  background-color: #06092b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  color: #fff;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
  font-weight: bold;
}

.c3 {
  font-size: 2rem;
  margin-bottom: 3rem;
  font-weight: 400;
}

.c4 {
  width: min(30rem,100%);
}

<div
  class="c0"
>
  <img
    alt="Imagem de um átomo e ao lado React Avançado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Pessoa em frente ao computador programando"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</div>
`
